cli/popper/commands/cmd_add.py

```diff
@@ -57,20 +58,25 @@ def cli(ctx, pipeline, folder, branch):
         os.mkdir(pipelines_dir)
 
     gh_url = 'https://github.com/{}/{}/'.format(owner, repo)
-    gh_url += 'archive/{}.zip'.format(branch)
+    gh_url += 'archive/{}.tar.gz'.format(branch)
 
     pu.info("Downloading pipeline {}... ".format(pipe_name))
 
-    r = requests.get(gh_url)
-    if r.status_code != 200:
-        pu.fail("Unable to fetch the pipeline. Please check if the name" +
-                " of the pipeline is correct and the internet is connected")
-    with zipfile.ZipFile(BytesIO(r.content)) as z:
-        z.extractall()
+    try:
+        file_tmp = urllib.request.urlretrieve(gh_url, filename=None)[0]
```

Comment:
can you comment (here on github) on why `requests` was replaced by `urllib`? Did you try changing `ZipFile` by `TarFile` (e.g. replacing `zipfile.ZipFile` by `tarfile.TarFile`) and see if that worked?